### PR TITLE
Refactor DateUtil and fix being able to add invalid dates

### LINKED ISSUE #209: Bug: Able to add dates that do not exist as appointment date or date of birth
* As the date format was `yyyy-MM-dd` , users can add invalid dates as long as they adhered to the format, e.g. `2024-09-31` which does not exist
* This is the case for 29 Feb such as being able to add `2023-02-29` even though that year was not a leap year

### PR DESCRIPTION
**Pull request by yong-zhuo**
fixes #209 

* Fix being able to add dates that do not exist like `2023-02-29` or `2024-09-31` by changing the `DATE_FORMAT` from `yyyy-MM-dd` to `uuuu-MM-dd`

* Also enabled a test case in `JsonSerializableAddressBook` that was previously disabled

* There is also 4 .md files in the `team` folder to indicate our contributions, we can choose to update those if we want in the future

### CODE REVIEW

File: src/main/java/seedu/address/commons/util/DateUtil.java [hunk 1]
```diff
@@ -3,40 +3,48 @@
...
+    }
+
+    private static Optional<LocalDate> parseDate(String date) {
+        try {
+            LocalDate parsedDate = LocalDate.parse(date, DATE_FORMATTER);
+            logger.fine("Parsed date: " + parsedDate);
+            return Optional.of(parsedDate);
+        } catch (DateTimeException e) {
+            logger.warning("Unable to parse date: " + date);
+            return Optional.empty();
```
Comment (hweinian) [inline src/main/java/seedu/address/commons/util/DateUtil.java:46]: good use of optional to maintain slap

**Review by hweinian — APPROVED** [review]

File: src/main/java/seedu/address/commons/util/DateUtil.java [hunk 2]
```diff
@@ -3,40 +3,48 @@
...
-     * @param date The date to be checked.
+     * Returns if a given string is a valid date in the correct format.
      */
     public static boolean isValidDate(String date) {
-        try {
-            LocalDate parsedDate = LocalDate.parse(date, DATE_FORMATTER);
-            return true;
-        } catch (DateTimeException e) {
-            return false;
-        }
```
Comment (TechRaj) [inline src/main/java/seedu/address/commons/util/DateUtil.java:26]: This SLAPs

File: src/test/java/seedu/address/commons/util/DateUtilTest.java [hunk 3]
```diff
@@ -28,25 +27,25 @@ public void isValidDate_invalidDate_returnsFalse() {
...
-        assertFalse(isAfterToday("2022-13-01")); // invalid month
-        assertFalse(isAfterToday("2022-12-32")); // invalid day
-        assertFalse(isAfterToday("2022/12/12")); // invalid format
+    public void isDateAfterToday_invalidDate_returnsFalse() {
+        assertFalse(isDateAfterToday("")); // empty string
+        assertFalse(isDateAfterToday(" ")); // spaces only
+        assertFalse(isDateAfterToday("example")); // non-numeric
+        assertFalse(isDateAfterToday("2022-13-01")); // invalid month
+        assertFalse(isDateAfterToday("2022-12-32")); // invalid day
+        assertFalse(isDateAfterToday("2022/12/12")); // invalid format
```
Comment (TechRaj) [inline src/test/java/seedu/address/commons/util/DateUtilTest.java:49]: Good tests

**Review by TechRaj — APPROVED** [review]
LGTM

**Pull request merged**